Jenkinsfile

```diff
@@ -1,12 +1,52 @@
-String cron_string = BRANCH_NAME == "master" ? "@hourly" : BRANCH_NAME.startsWith("PR") ? "@midnight" : ""
+// Get current date
+def now = new Date()
+
+// isTriggerByTimer == true if the timer triggers it
+// isEmpty() on lists is currently broken in Jenkins pipelines... so we have to rely on the size of the list
```

Comment:
Is there a Jenkins ticket to follow this?

Reply:
The root of it is actually: https://issues.jenkins.io/browse/JENKINS-50863 and at some point list.isEmpty() was removed from the  default whitelist. I was following the issue a few years ago myself but gave up on it.